### ISSUE TITLE
Gemma Tokenizer add padding to the left, but the code to find sequence length assume pad tokens are added to the right

### ISSUE THREAD

**Issue by binh-vu** [huggingface/transformers#30004]
Gemma Tokenizer add padding to the left, but the code to find sequence length assume pad tokens are added to the right

Reference:
https://github.com/huggingface/transformers/blob/97c00cdfe132164dbd793447a088432fa359fd36/src/transformers/models/gemma/modeling_gemma.py#L1318-L1320

**Comment by jinseriouspark** [comment]
Hello! What was your conclusion on this topic? Since you completed it without any additional comments from others, I'm leaving a comment out of curiosity! Did you fine tune it by changing the padding to the right side?

**Comment by tsor13** [comment]
I ran into the same issue, I was able to successfully finetune by changing the tokenizer to pad right! With the left padding, I was getting nan logits out of gemma2

**Comment by Taimin** [comment]
> I ran into the same issue, I was able to successfully finetune by changing the tokenizer to pad right! With the left padding, I was getting nan logits out of gemma2

Similar here. I got nan grad norm during training. anyone know the reason?
